rules/library.bzl

```diff
@@ -591,9 +591,11 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
     framework_vfs_overlay_name = name + "_vfs"
     framework_vfs_overlay_name_swift = swift_libname + "_vfs"
 
-    # TODO: remove framework if set
+    # TODO: remove under certian circumstances when framework if set
```

Comment:
when framework if set?

Reply:
It's not clear to me what the scenarios are - and the original comment doesn't really indicate when.

Reply:
I've reverted it to the original logic as it worked and I'm ignorant of it 
> This PR reverts to the orginal, albeit more nuanced logic.